Add intro blog post (#13)
skodjob/blog: posts/general/new-boy-in-town.md

---

title: Welcome this little boy to the town
menu_order: 1
post_status: publish
post_excerpt: An introduction about us, our work and what we want to do in scope of Skodjob.io
featured_image: _images/skodjob.jpg
author: jstejskal
taxonomy:
    category:
        - general

---

# Welcome this little boy to the town

> The whole idea started a few years back. Our main work consists of testing system, that are running on top of Kubernetes. 

Reviewer: ```suggestion
The whole idea started a few years back. Our main work consists of finding all sorts of defaults in applications running on top of Kubernetes.
```

That itself is quite a challenge if we consider that systems like that could be distributed systems, integration of several different components, or even integration of several systems together.

> Our experience was mostly about spinning up Kubernetes cluster, deploy our project, in our case [Strimzi](https://strimzi.io/), play with it a little bit on test level, and then evaluate the state with some result.

Reviewer: ```suggestion
Our experience was mostly about spinning up the Kubernetes cluster, deploying our project, in our case, [Strimzi](https://strimzi.io/), playing with it a little bit on the test level, and then evaluating the state with some results.
```

> That was enough for testing most of the features we introduced. However, at some point it started to be a little bit boring so we were thinking about some new challenge.

Reviewer: ```suggestion
That was enough for testing most of the features we introduced. However, at some point, it started to be a little bit boring so we were thinking about some new challenge.
```

## Codename Teal'c

> During Covid-19 pandemic we spent a lot of time home and during that time one of the best TV series was online - Stargate SG-1.

Reviewer: ```suggestion
During the Covid-19 pandemic, we spent a lot of time at home and during that time one of the best TV series was online - Stargate SG-1.
```

> No worries, I am not going to promote the series or spoils the plot to you! 

Reviewer: ```suggestion
No worries, I am not going to promote the series or spoil the plot for you! 
```

> However, within our team we liked Teal'c character, and then it started - let's start to work on something and le'ts called it Teal'c!

Reviewer: ```suggestion
However, within our team, we liked Teal'c character, and then it started - let's start to work on something, and let's call it Teal'c!
```

> We started with an initiative that's goal was to test our projects more like customers.

Reviewer: ```suggestion
We started with the idea to test our project in an environment that is more or less similar to what customers could be using.
```

> Our existing approach was fine for verifying features, but it was usually done in small scale. 

Reviewer: ```suggestion
Our existing approach was fine for verifying features, but it was usually done in a small scale. 
```

> A lot of questions suddenly comes to our minds like:

Reviewer: ```suggestion
A lot of questions suddenly come to our minds like:
```

> - how the software be behave when it will running for several weeks/months?

Reviewer: ```suggestion
- how does the software behave when it will run for several weeks/months?
```

- how we can easily automate the deployment and upgrade process?
- how upgrades of underlying architecture influences it?
- actually, how users are using it? :O
- what about monitoring?
- collecting logs?
- chaos?!

> So we took that road and during free time we moved from regular verification to customer simulation with a lot of unknown.

Reviewer: ```suggestion
So we took that road and during free time we moved from regular verification to customer simulation with a lot of unknowns.
```

## Teal'c is dead, long live Skodjob
It took some time, but we started to reap the fruits of our labor. 
From quite complex (but still simple) testing we moved to continuous verification of several projects that were deployed for months!
Issues were found, motivation enlarged, and experience? Marvelous.

> During the years we did several talks, write some blog posts about our work, and learn a lot of cool stuff. 

Reviewer: ```suggestion
During the years we did several talks, wrote some blog posts about our work, and learned a lot of cool stuff. 
```

However, we were still missing something. 
The freedom of sharing our knowledge quickly and easily somewhere, where we could manage it on our own.
We wanted to write down our ideas and experience after a couple of years of experience without any limitations.

And here we are, on `skodjob.io`. 

> It is good feeling, and we all hope, that you will enjoy our posts where we will try to provide as much information as we may, and also information that we were missing in other posts that we came across in the past.

Reviewer: ```suggestion
It is a good feeling, and we all hope, that you will enjoy our posts where we will try to provide as much information as we may, and also information that we were missing in other posts that we came across in the past.
```
